near-contract-standards/src/non_fungible_token/metadata.rs

```diff
@@ -0,0 +1,62 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::json_types::Base64VecU8;
+use near_sdk::serde::{Deserialize, Serialize};
+
+pub const NFT_METADATA_SPEC: &str = "nft-1.0.0";
+
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(crate = "near_sdk::serde")]
+pub struct NFTContractMetadata {
+    pub spec: String,              // required, essentially a version like "nft-1.0.0"
+    pub name: String,              // required, ex. "Mosaics"
+    pub symbol: String,            // required, ex. "MOSIAC"
+    pub icon: Option<String>,      // Data URL
+    pub base_uri: Option<String>, // Centralized gateway known to have reliable access to decentralized storage assets referenced by `reference` or `media` URLs
+    pub reference: Option<String>, // URL to a JSON file with more info
+    pub reference_hash: Option<Base64VecU8>, // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, BorshDeserialize, BorshSerialize)]
+#[serde(crate = "near_sdk::serde")]
+pub struct TokenMetadata {
+    pub title: Option<String>, // ex. "Arch Nemesis: Mail Carrier" or "Parcel #5055"
+    pub description: Option<String>, // free-form description
+    pub media: Option<String>, // URL to associated media, preferably to decentralized, content-addressed storage
+    pub media_hash: Option<Base64VecU8>, // Base64-encoded sha256 hash of content referenced by the `media` field. Required if `media` is included.
+    pub copies: Option<u64>, // number of copies of this set of metadata in existence when token was minted.
+    pub issued_at: Option<String>, // ISO 8601 datetime when token was issued or minted
+    pub expires_at: Option<String>, // ISO 8601 datetime when token expires
+    pub starts_at: Option<String>, // ISO 8601 datetime when token starts being valid
+    pub updated_at: Option<String>, // ISO 8601 datetime when token was last updated
+    pub extra: Option<String>, // anything extra the NFT wants to store on-chain. Can be stringified JSON.
+    pub reference: Option<String>, // URL to an off-chain JSON file with more info.
+    pub reference_hash: Option<Base64VecU8>, // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
```

Comment:
I know this might be an annoying comment, but switching those comments to doc comments and adding a quick description of the two structures would be useful imo

Reply:
Totally. I got ambitious just now thinking I could Comment All The Things™ and realize this is actually quite a task to do. I'm hoping our whole team can lean into this soon.